sklearn/mixture/bayesianmixture.py

```diff
@@ -0,0 +1,985 @@
+"""Bayesian Gaussian Mixture Model."""
+import numpy as np
+from scipy import linalg
+from scipy.special import digamma, gammaln
+
+from ..utils import check_array
+from ..utils.validation import check_is_fitted
+from .base import MixtureBase, check_shape, check_X
+from .gaussianmixture import estimate_Gaussian_suffstat
+
+
+def xlogx(x):
+    return np.sum(x[x < np.e]) + np.sum(x[x > np.e])
```

Comment:
Sorry, this was used for testing the underflow problem. I accidentally committed.